custom_components/dahua/client.py

```diff
@@ -77,7 +85,13 @@ async def async_get_system_info(self) -> dict:
         updateSerial=IPC-HDW5830R-Z
         updateSerialCloudUpgrade=IPC-HDW5830R-Z:07:01:08:70:52:00:09:0E:03:00:04:8F0:00:00:00:00:00:02:00:00:600
         """
-        return await self.get("/cgi-bin/magicBox.cgi?action=getSystemInfo")
+        not_hashed_id = "{0}_{1}_{2}_{3}".format(self._address, self._rtsp_port, self._username, self._password)
+        unique_cam_id = md5((not_hashed_id).encode('UTF-8')).hexdigest()
```

Comment:
Seems this work should only be done in the "except" case. Otherwise it's done and thrown away for nothing.

Reply:
Yes it is. For RTSP need to make json structure that can be used by "class DahuaDataUpdateCoordinator" in file "__init__.py"